pkg/util/controller_finder.go

```diff
@@ -456,3 +547,75 @@ func verifyGroupKind(ref *rolloutv1beta1.ObjectRef, expectedKind string, expecte
 
 	return false, nil
 }
+
+// getControllerRevisionsForDaemonSet returns ControllerRevisions for the given DaemonSet,
+// sorted by revision number in descending order (newest first)
+func (r *ControllerFinder) getControllerRevisionsForDaemonSet(daemonSet *apps.DaemonSet) ([]*apps.ControllerRevision, error) {
+	// List all ControllerRevisions in the same namespace
+	revisionList := &apps.ControllerRevisionList{}
+
+	err := r.List(context.TODO(), revisionList, utilclient.DisableDeepCopy,
+		&client.ListOptions{Namespace: daemonSet.Namespace})
+	if err != nil {
+		return nil, err
+	}
+
+	var revisions []*apps.ControllerRevision
+	for i := range revisionList.Items {
+		revision := &revisionList.Items[i]
+		// Check if this ControllerRevision is owned by the DaemonSet
+		if ref := metav1.GetControllerOf(revision); ref != nil {
+			if ref.Name == daemonSet.Name && ref.Kind == "DaemonSet" {
```

Comment:
please also check the apiversion field of the reference. Alternatively, just check the uid of the ref ?